src/link/ping360simulationlink.cpp

```diff
@@ -66,11 +68,27 @@ void Ping360SimulationLink::randomUpdate()
     // Calculate the global average time between requests
     _counter++;
     float elapsedTime = _elapsedTimer.elapsed();
-    if (_counter == 1) {
-        _globalAverageTimeMs = elapsedTime;
-    } else {
-        float weight = (_counter - 1) / static_cast<float>(_counter);
-        _globalAverageTimeMs = _globalAverageTimeMs * weight + elapsedTime * (1 - weight);
-    }
+    float weight = (_counter - 1) / static_cast<float>(_counter);
+    _globalAverageTimeMs = _globalAverageTimeMs * weight + elapsedTime * (1 - weight);
+
     _elapsedTimer.restart();
+
+    // This should be done after _counter increase to not interate before a full first spin
+    if (_counter % angularResolution == 0) {
+        _spins++;
+    }
+
+#if defined(PING360_SPEED_TEST)
```

Comment:
simulationlink seems like the wrong place to perform all this elapsed time calculations and code for the test.

I would put this in sensor like Ping360::doFullScanSpeedTest().

Reply:
This will not simulate the link connection and everything related to the latency between sensor and the abstractlink logic.

Reply:
I agree that is not **the best** place to have it, but we can improve it later.